### PR TITLE
HAL-1578: Customise Browser Title

### PR DESCRIPTION
**Pull request by hpehl**
https://issues.jboss.org/browse/HAL-1578

### CODE REVIEW

**Comment by hpehl** [comment]
Reopened using upstream issues from #195

**Comment by honza-kasik** [comment]
Continuing previous discussion: 

I am just thinking if migration of such settings into "console subsystem" won't it be a problem for users? Settings won't be "per-user" anymore but rather "per-instance". What are the obstacles in introducing that subsystem now?

**Comment by hpehl** [comment]
The introduction of a console subsystem needs more analysis: What are "per-user" settings, what are "per-instance" settings. I think this won't happen for WF17.

**Comment by honza-kasik** [comment]
Oh, sorry, I meant "per-user" as in that each browser has one user, and using cookies to save title makes Web Console title unique for each user. On the other hand, introducing some server side storage would make title available for all users on given instance.

File: console/HAL-1578_Customise_Browser_Title.adoc [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+:issue-base-url:    https://issues.jboss.org/browse
+
+== Overview
+
+Customise the title of the management console. A custom title can be specified in the user settings of the console. The title can contain the following placeholders:
+
+- `%n`: The `name` attribute of the the root resource
+- `%o`: The `organization` (standalone) resp. `domain-organization` attribute of the the root resource
+
+If the title is not specified, the built in title will be used. If a placeholder is used, but the attribute is undefined, the built in title will be used.
```
Comment (honza-kasik) [inline console/HAL-1578_Customise_Browser_Title.adoc]: Wouldn't it be better if users could directly benefit from the enhancement by using the name placeholder in default title value?

Comment (hpehl) [reply console/HAL-1578_Customise_Browser_Title.adoc]: AFAIK the name attribute is undefined for domain mode.

Comment (honza-kasik) [reply console/HAL-1578_Customise_Browser_Title.adoc]: It seems like it has a name, though not a nice one:
```
[domain@localhost:9990 /] :read-attribute(name=name)
{
    "outcome" => "success",
    "result" => "Unnamed Domain"
}
```

**Comment by honza-kasik** [comment]
@hpehl To conclude that discussion. You don't think, that in future, when console subsystem will be introduced and it will be used to store the title value, the behavior (per-user vs per-instance) will not be changed/will not make any problems to users?

**Comment by honza-kasik** [comment]
@hpehl Can you please also add information about who does the testing (QA in this case)?

**Comment by hpehl** [comment]
> @hpehl To conclude that discussion. You don't think, that in future, when console subsystem will be introduced and it will be used to store the title value, the behavior (per-user vs per-instance) will not be changed/will not make any problems to users?

I don't know whether or when a console subsystem will be introduced. If so I'm sure we'll find a smooth migration for the settings. 

> @hpehl Can you please also add information about who does the testing (QA in this case)?

You mean in the analysis document?

**Comment by honza-kasik** [comment]
> You mean in the analysis document?

Yes, please - similar to this: https://github.com/wildfly/wildfly-proposals/pull/196/files#diff-09c496d673c56836551e84041b44a56cR32

**Comment by hpehl** [comment]
> > You mean in the analysis document?
> 
> Yes, please - similar to this: https://github.com/wildfly/wildfly-proposals/pull/196/files#diff-09c496d673c56836551e84041b44a56cR32

Done.

File: console/HAL-1578_Customise_Browser_Title.adoc (hunk 1 above)
Comment (honza-kasik) [reply console/HAL-1578_Customise_Browser_Title.adoc]: @hpehl I understand there will be no `%n` in title by default? IMHO it would make sense...

Comment (hpehl) [reply console/HAL-1578_Customise_Browser_Title.adoc]: Ok, I agree. I'll change the implementation: Default title will be "%n | Management Console"

Comment (honza-kasik) [reply console/HAL-1578_Customise_Browser_Title.adoc]: Thanks!

**Comment by hpehl** [comment]
@honza-kasik squashed to one commit.

**Pull request merged**